### ISSUE TITLE
Stylized UI for filtering by project type

### ISSUE THREAD

**Issue by ohsk** [urbanlaunchpad/SoloCityMap#46]
See #23 for previous explorations.

Other references:
http://tympanus.net/Development/SelectInspiration/index2.html

**Comment by ohsk** [comment]
@buzoherbert I've been brainstorming about the filter by project type. I kind of want to take it in a very different direction than we've been talking about so far. Initial mockup is below.

So the basic idea is to present the distribution of project types visually with a bar, subdivided based on the project cost by each type. Then, the user could mouse over that bar and see the amounts for specific project types while also getting a sense of where that Kelurahan is budgeting by each type. The bar acts as the filter mechanism itself.

Phase | Exhibit
--- | ---
Kelurahan selected | ![image](https://cloud.githubusercontent.com/assets/1689222/5989411/382549f6-a93c-11e4-9b4d-c23bc63caa8c.png)
Mouseover | ![image](https://cloud.githubusercontent.com/assets/1689222/5989435/094a838e-a93d-11e4-9694-8c60607db85a.png)
Project type selected from bar | ![image](https://cloud.githubusercontent.com/assets/1689222/5989444/3d609d66-a93d-11e4-9df5-ced9e47ceab5.png)

I think it also helps with the fact that not every Kelurahan has each project type. It's kind of annoying to filter by a project type and not see anything just because none of those exist for that Kelurahan. This way, you are really only interacting with the project types present.

Do you think this is doable??

**Comment by ohsk** [comment]
([Inspired by this NYT project](http://www.nytimes.com/interactive/2014/09/03/us/the-race-gap-in-americas-police-departments.html?_r=0))

**Comment by buzoherbert** [comment]
Hey!
Yeah, that's totally doable with our database. So, on click on any of the colors you would see the list of projects of that kind?

It's a very good idea. Maybe we can implement something like that for other info, like ghost projects.

I just wonder how well will it work when there's a huge difference between the butgets on one subcategory versus another one...

**Comment by ohsk** [comment]
Yeah, that's a good point when there's a huge difference. I'd say let's try it and see how things look! Maybe we'd have to have a minimum width (1px) for any color bar or something...the NYT's version seems to work well even with really small values:

![image](https://cloud.githubusercontent.com/assets/1689222/5992707/ef853a00-a9e8-11e4-984a-6956c2ba7e27.png)

And yeah, I think we could add a filter right above or below that allows you to (de)select ghost vs. voted projects, like on this [Urban Institute](http://datatools.urban.org/features/mapping-americas-futures/) map:

![image](https://cloud.githubusercontent.com/assets/1689222/5992709/1d7f4e5a-a9e9-11e4-8939-25b1e8f95901.png)

So something like:

 Project Origin |  |  |
--- | --- | ---
All | Voted | Ghost

 Project Status | |  |
--- | --- | ---
All | Completed | Incomplete
